app/models/manageiq/providers/ovirt4/infra_manager.rb

```diff
@@ -0,0 +1,29 @@
+require 'ovirtsdk4'
+
+class ManageIQ::Providers::Ovirt4::InfraManager < ManageIQ::Providers::InfraManager
```

Comment:
Why do you introduce a new provider?
I'd rather put a feature toggle into the connect method in the current `InfraManager`

I did something similar when moving to aws sdk v2.
I put it into raw_connect: https://github.com/ManageIQ/manageiq/pull/6297/files#diff-235184baadf210fc9a4d037d8ffa1c37